### PR TITLE
Llc ssl kin curl

### PR DESCRIPTION
**Pull request by karlschwarz**
Pull request of llc_ssl_kin_curl branch. : )

### CODE REVIEW

File: ulmo/llc/io.py [hunk 1]
```diff
@@ -86,6 +86,13 @@ def grab_llc_datafile(datetime=None, root='LLC4320_', chk=True, local=False):
     # Return
     return datafile
                     
+def load_llc_ds(filename, local=False):
```
Comment (profxj) [inline ulmo/llc/io.py:89]: add doc string (I can do it)

File: ulmo/runs/SSL/llc_ssl_kin/llc_ssl_kin.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
+from __future__ import print_function
```
Comment (profxj) [inline ulmo/runs/SSL/llc_ssl_kin/llc_ssl_kin.py:1]: It looks like we have two `llc_ssl_kin.py` files.

Can you merge them?  I think the final file
should go in SSL/LLC

File: ulmo/runs/SSL/llc_ssl_kin/llc_ssl_kin.py [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+    train_loader = modis_loader_v2(opt)
+
+    # build model and criterion
+    model, criterion = set_model(opt, cuda_use=opt.cuda_use)
+
+    # build optimizer
+    optimizer = set_optimizer(opt, model)
+    
+    # read 'user' and 'pin' for comet log
+    with open('/etc/comet-pin-volume/username', 'r') as f:
```
Comment (profxj) [inline ulmo/runs/SSL/llc_ssl_kin/llc_ssl_kin.py:61]: what is `comet`?

Going forth, let's use the `main_train` code in the `modis_ssl_full` branch

File: ulmo/runs/SSL/llc_ssl_kin/llc_ssl_kin.py [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+        modis_data: (numpy.array) modis_data used in the latents
+            extraction process.
+        model_path: (string) path of the model file. 
+        save_path: (string) path to save the extracted latents
+        save_key: (string) key of the h5py file for the latents.
+    """
+    model, _ = set_model(opt, cuda_use=opt.cuda_use)
+    model_dict = torch.load(model_path)
+    model.load_state_dict(model_dict['model'])
+    modis_data = np.repeat(modis_data, 3, axis=1)
```
Comment (profxj) [inline ulmo/runs/SSL/llc_ssl_kin/llc_ssl_kin.py:114]: This is a bad use of memory.

See how I am doing it now in `modis_ssl_full`

Comment (profxj) [reply ulmo/runs/SSL/llc_ssl_kin/llc_ssl_kin.py:114]: But we can keep as is here.

**Review by profxj — CHANGES_REQUESTED** [review]
A few things to work on .

Thanks!

File: ulmo/runs/SSL/llc_ssl_kin/llc_ssl_kin.py (hunk 2 above)
Comment (karlschwarz) [reply ulmo/runs/SSL/llc_ssl_kin/llc_ssl_kin.py:1]: Sure, start to do it now.

**Review by profxj — APPROVED** [review]
all set on my end

File: ulmo/runs/SSL/llc_ssl_kin/llc_ssl_kin.py (hunk 3 above)
Comment (karlschwarz) [reply ulmo/runs/SSL/llc_ssl_kin/llc_ssl_kin.py:61]: Comet is used to show the training process, namely the learning curve. We comment out the codes related to the comet in the main file.

File: ulmo/runs/SSL/llc_ssl_kin/llc_ssl_kin.py (hunk 4 above)
Comment (karlschwarz) [reply ulmo/runs/SSL/llc_ssl_kin/llc_ssl_kin.py:114]: Yes, here we replaced the old function with the model_latents_extract function from latents_extraction file.

File: ulmo/llc/io.py (hunk 1 above)
Comment (karlschwarz) [reply ulmo/llc/io.py:89]: Yes, it is done.

**Pull request merged**